src/main/java/com/hierynomus/msdfsc/ReferralCache.java

```diff
@@ -87,6 +87,11 @@ public ReferralCacheEntry lookup(DFSPath dfsPath) {
         return referralEntry;
     }
 
+    public void clear(DFSPath dfsPath) {
+        List<String> pathComponents = dfsPath.getPathComponents();
+        cacheRoot.deleteExpiredReferralEntry(pathComponents.iterator());
```

Comment:
instead of sending iterator, wouldn't it be a little cleaner to just to send the `List`?

Reply:
It's recursive and the other functions used an iterator; changed it to Lists anyway.